Module improvements

josephco opened the pull request:
This is tied to https://github.com/CasperSleep/nightshade/pull/61, which should be merged after this

josephco committed:
Create basic Gallery and rename old gallery files

josephco committed:
Gallery and quote style tweaks

josephco committed:
Add option for custom imgix class to panel images

josephco committed:
Create ScrollTo.js smooth scroll animation util

misscs commented on src/content_panel/panel.macros.html:
```diff
@@ -5,10 +5,10 @@
   * Renders a content and image panel.
 #}
-{% macro content_panel(heading, image_path, image_description=null, image_position='left', class=null, id=null) %}
+{% macro content_panel(heading, image_path, image_description=null, image_position='left', imgix_class=null, class=null, id=null) %}
```
I dig `imgix_class` as the standard param key for any module that has media coming from imgix.

misscs commented on src/animation/ScrollTo.js:
```diff
@@ -0,0 +1,57 @@
+      window.scrollTo(0, this.targetPosition);
+    }
+  }
+};
```
This is a tiny enough module to refactor to provide a closure for the methods. Then they'll be "private" :mag: 

```
export const modulename = {

// private method here, no need to use underscore prefix

// another private

return {
 // return scroll()
}


};

```

josephco replied:
I tried doing it this way, but ran into some JS errors. It seems it has to be defined as a self-invoking function and rethought somewhat. I'd like to come back to this after talking with Rob.

josephco committed:
Correctly apply imgix-fluid behavior to images

Use data-src for background images on heros/ctas
Use --nocss and instead have image SCSS in modules

josephco committed:
CTA and Hero tweaks

josephco committed:
Fix IE JavaScript errors

josephco committed:
Move IE9 styles to module specific partials
